### DISCUSSION TITLE
Detector acquisition rate changes depending on mouse movement

### DISCUSSION THREAD

**Discussion by rgeneaux** [PyMoDAQ/PyMoDAQ#109] [Q&A]
Hello everyone,

I was working on implementing frame rate readings to viewers and turned into a funny behavior when testing. The code is quite simple (here: https://github.com/rgeneaux/PyMoDAQ/tree/fps_dev). First I made sure that the measured frame rate does not depend on the "refresh rate" parameter, only on how quick data is acquired.

Let's consider the DAQ0D Mock, which has a "wait time" attribute (which would be the exposure time of a real device). 
If you set the wait time to 1 ms, you get an FPS of 64 Hz, much smaller than the theoretical 1 kHz.
But the stranger part is that if you move your mouse back and forth on the interface, the FPS goes up to >500 Hz! As if hovering the elements of the interface forced an acquisition or something.

Here it is in video: (with and without the FPS reading activated)

https://user-images.githubusercontent.com/295505/229863631-4eb5f9b9-3066-4862-88d7-e9c6f83646a3.mov

It is less noticeable at lower speeds but still there: with wait time = 100 ms, FPS = 9.2 Hz without moving, and 9.8-9.9 Hz with moving.

Any idea of what causes that behavior?

Romain

**Comment by rgeneaux** [comment]
Hi,
Bumping this question as I am seeing something similar with an Andor CMOS camera. 
Somehow moving the mouse allows to read frames faster out of the camera which is quite strange! @seb5g have you observed something similar?

**Reply by seb5g** [reply]
nope never see that. It seems to be related to qt forcing the refreshing?!

Could you do your test using the plugin live mode? And then put a wait time within your plugin to see if the same behaviour happens. The MockCamera 2D plugin has this functionnality, I'd like to see if it is similar
